dts/bindings/flash_controller/nxp,iap-mcx.yaml

```diff
@@ -0,0 +1,8 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP MCX Flash Memory Module (FMU)
+
+compatible: "nxp,iap-mcx"
```

Comment:
Could we avoid the family binding in the flash controller name here?

Reply:
The binding name was picked after looking at the SDK driver used for this platform. I

Reply:
Updated the binding name to be more generic